superset-frontend/src/components/AlteredSliceTag/index.jsx

```diff
@@ -147,15 +147,12 @@ export default class AlteredSliceTag extends React.Component {
     if (controlsMap[key]?.type === 'CollectionControl') {
       return value.map(v => safeStringify(v)).join(', ');
     }
-    if (controlsMap[key]?.type === 'MetricsControl' && Array.isArray(value)) {
-      const formattedValue = value.map(v => (v.label ? v.label : v));
-      return formattedValue.length ? formattedValue.join(', ') : '[]';
-    }
     if (typeof value === 'boolean') {
       return value ? 'true' : 'false';
     }
     if (value.constructor === Array) {
-      return value.length ? value.join(', ') : '[]';
+      const formattedValue = value.map(v => (v.label ? v.label : v));
```

Comment:
This could potentially be called for other control types aside from metrics.
Why did you move it from above? Was the code not entering? We're essentially switching from `Array.isArray` to `value.constructor`, which should have similar effect.
If that's the only reason, we could replace `Array.isArray` with `value.constructor` in the other block you removed, but keeping it locked to the metrics control.

Reply:
@diegomedina248 
It does make sense. Resolved.